Support offices page

Mahmoud-Emad opened the pull request:
### Description

This feat is a part of https://github.com/codescalers/CSHR/issues/433#issuecomment-2213415297 

### Changes

- Reorder the SideBar
- Created a new component to serve 404 requests
- Created a new component to serve the office view
- Request to load the office holidays based on the selected day
- Implement a new endpoint to list office holidays
- Support pagination to list paginated offices

### Related Issues

- https://github.com/codescalers/CSHR/issues/441

### Screenshots

- ![image](https://github.com/user-attachments/assets/29742e7f-1948-42fc-bacd-4020436358f8)
- ![image](https://github.com/user-attachments/assets/056fbcf8-8a0e-43ee-ac91-d9c2973b82b1)

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

zaelgohary commented on client/src/components/dialogs/OfficeHolidaysDialog.vue:
```diff
@@ -0,0 +1,146 @@
+        }}</v-alert>
+        <div class="d-flex flex-row">
+          <v-tabs v-model="tab" color="primary" direction="vertical">
+            <!-- <v-text-field
+            style="max-width: 150px;"
+            v-model="requestYears"
+            variant="underlined"
+            label="Last years"
+            color="primary"
+            :rules="[
+              (v: string) => !isNaN(+v) || 'Please enter only numeric values in this field.',
+              (v: string) => +v <= 7 || 'The value should not be more than 7.',
+              (v: string) => +v >= 1 || 'The value should not be less than 1.'
+            ]"
+          /> -->
```
Please remove the commented code.

mahmoud-emad replied:
This feature is hidden, but we may need it in the future.

zaelgohary commented on server/cshr/serializers/public_holidays.py:
```diff
@@ -16,3 +16,14 @@ class Meta:
+
+    class Meta:
+        model = PublicHoliday
+        fields = ["holiday_date", ]
```
```suggestion
        fields = ["holiday_date"]
```

mahmoud-emad replied:
This shouldn't be done. If I applied your suggestion, the Python interpreter will read it as a `string` type, while it should be a `tuple` type.

zaelgohary commented:
I navigated to wrong route and 404 page did not show.

![image](https://github.com/user-attachments/assets/00362012-1bf6-479a-b514-d68d8d3ac7bf)

zaelgohary commented:
We started using CSHR in 2024 and have no data prior to this year. Do we still need to show empty data for the past four years? I suggest removing them.

I also don't think we need the line `Filtering the total public holidays based on the year 2024.` as it's self explanatory that filtering will be done based on the chosen year.

Maybe we can add a margin left to `Egypt Office` to be aligned with the rest of the content?


![image](https://github.com/user-attachments/assets/9725b2ed-636b-476d-b6d1-378cc3784ed1)

Mahmoud-Emad commented:
> I navigated to wrong route and 404 page did not show. 404 route should be added to the router.
> 
> ![image](https://private-user-images.githubusercontent.com/40770501/365435416-00362012-1bf6-479a-b514-d68d8d3ac7bf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxMjc3OTQsIm5iZiI6MTcyODEyNzQ5NCwicGF0aCI6Ii80MDc3MDUwMS8zNjU0MzU0MTYtMDAzNjIwMTItMWJmNi00NzlhLWI1MTQtZDY4ZDhkM2FjN2JmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA1VDExMjQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMzhiODZkMGMxOTRlZjExNmVjN2QwOTNlZjA2MzI5NjUyMGM4NWFjOGZhZGYxZWRjNDM1ZDYyMTQ5NDdmYTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AiK2vV7hh7H_HhcCCi6vwmcj3K9kj0tIdCIvdNjJDoA)

Please open an issue to support this

Mahmoud-Emad commented:
> We started using CSHR in 2024 and have no data prior to this year. Do we still need to show empty data for the past four years? I suggest removing them.
> 
> I also don't think we need the line `Filtering the total public holidays based on the year 2024.` as it's self-explanatory that filtering will be done based on the chosen year.
> 
> Maybe we can add a margin left to `Egypt Office` to be aligned with the rest of the content?
> 
> `public holidays` needs to be capitalized as well.
> 
> ![image](https://private-user-images.githubusercontent.com/40770501/365435591-9725b2ed-636b-476d-b6d1-378cc3784ed1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxMjc3OTQsIm5iZiI6MTcyODEyNzQ5NCwicGF0aCI6Ii80MDc3MDUwMS8zNjU0MzU1OTEtOTcyNWIyZWQtNjM2Yi00NzZkLWI2ZDEtMzc4Y2MzNzg0ZWQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA1VDExMjQ1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjNzY1ZjJmMzZlYTk1OTRlM2I3MzRlMTkzNzkyNzY4MjEzY2JkNjhkYzUxOTA1YjQxNGYyYWU0NmYwNmRhMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3frthNlL5Nn25hgEd2mZrtlZw5OK14qs0HFeKd5HqME)

1. The idea about the first point is something like future planning, now we return to the last 5 years, yeah we don't have data now but think about it what will happen next year? and the year after it as well and so on.

2. Removed

3. Removed the card title since we have an alert with the needed info, so it doesn't matter to have the card title component

3.
